app/controllers/settings_controller.py

```diff
@@ -1044,6 +1052,20 @@ def _on_community_rules_db_local_file_choose_button_clicked(self) -> None:
         )
         self._last_file_dialog_path = Path(community_rules_db_location).parent
 
+    @Slot()
+    def _on_community_rules_db_location_open_button_clicked(self) -> None:
+        """
+        Open the community rules database Folder Location.
+        """
+        community_rules_db_location = (
+            self.settings.external_community_rules_file_path.replace(
+                "communityRules.json", "dbs/Community-Rules-Database"
+            )
```

Comment:
For the DB location, it seems you are determining the parent folder via assuming there is a `communityRules.json`. But, isn't it possible for the user to choose a `.json` file that isn't named `communityRules`? In that case, there are community rules but this string replacement logic doesn't work as intended.

To find the db location I would use path logic instead using `os`.

Reply:
ok i will do a separate pull for this